### ISSUE TITLE
OS X El Captain HHVM 3.15.2 pgsql not loaded!

### ISSUE THREAD

**Issue by ahmadazimi** [facebook/hhvm#7475]
### HHVM Version
3.15.2
### Standalone code, or other way to reproduce the problem
get_loaded_extensions()
### Expected result
`pgsql` and `pdo_pgsql` should load automatically based on docs for this version!
### Actual result
`pgsql` and `pdo_pgsql` are not loaded!

Hi there, I've recently installed **hhvn 3.15.2** via homebrew on **OS X El Captain**.
But when I print `get_loaded_extensions()` output no result found for pgsql and pdo_pgsql.

Installed by this commands:
```
brew tap hhvm/hhvm
brew install hhvm
```

Should I do some extra things?

**Comment by Orvid** [comment]
The pgsql development headers need to be available when building HHVM, and that includes for OSX.

**Comment by ahmadazimi** [comment]
How can check if pgsql development headers available?

BTW:
I've installed **postgresql** using homebrew before building hhvm:
```
brew install postgresql
``` 

but pgsql not loaded.

**Comment by Orvid** [comment]
@simonwelsh - Is there anything special that has to be done to support it on OSX?

**Comment by simonwelsh** [comment]
Probably need to update the homebrew file to set the postgres paths if it's installed.

**Comment by ahmadazimi** [comment]
@simonwelsh I've upgrade postgres just before building hhvm, and also my postgres works in other environments.

```
postgresql: stable 9.5.5 (bottled), HEAD
```

**Comment by ahmadazimi** [comment]
I guess I've found the problem!
Please check [my comment ](https://github.com/facebook/hhvm/issues/7488#issuecomment-260139864)on issue #7488

**Comment by simonwelsh** [comment]
Homebrew installs the postgres headers in a location that is searched. The HHVM formula then explicitly changes where searching happens to just `/tmp`. The [hhvm-homebrew formula](https://github.com/hhvm/homebrew-hhvm/blob/master/hhvm.rb) will need to be updated to optionally set the postgres include dir and library locations.

**Comment by zerodiv** [comment]
Edit the hhvm brew formula to include:

1) $brew edit hhvm 

2) Add this around line 110 with the rest of the batch flag ads

# JEO: Added pgsql, ghetto enablement needs a ifstatement to check if the lib is around
cmake_args << "-DENABLE_EXTENSION_PGSQL=ON"
cmake_args << "-DPGSQL_INCLUDE_DIR=#{Formula["postgresql"].opt_include}"
cmake_args << "-DPGSQL_LIBRARY=#{Formula["postgresql"].opt_lib}"

3) brew install hhvm

Fails to compile with:
==> cmake -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/hhvm/3.17.2 -DDEFAULT_CONFIG_DIR=/usr/local/etc/hhvm -DCMAKE_CXX_COMPILER=/usr/local/opt/llvm/bin/clang++ -DCMAKE_C_COMPILER=/usr/local/opt/llvm/bin/clan
==> make
Last 15 lines from /Users/jorcutt/Library/Logs/Homebrew/hhvm/02.make:
                ^~~
                {  }
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/ref-data.h:255:17: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    m_hdr.init({0,0}, HeaderKind::Ref, 1); // cow=z=0, count=1
                ^~~
                {  }
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/ext/process/ext_process.cpp:484:23: warning: comparison of constant -1 with expression of type 'const bool' is always false [-Wtautological-constant-out-of-range-compare]
    } else if (result == -1) {
               ~~~~~~ ^  ~~

Actual issue looks to be:
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/ext/pgsql/pdo_pgsql_connection.cpp:246:12: error: ambiguous conversion for functional-style cast from 'long' to 'HPHP::String'
    return String((long)this->pgoid);
           ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:135:18: note: candidate constructor
  /* implicit */ String(int     n);
                 ^
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:136:18: note: candidate constructor
  /* implicit */ String(int64_t n);
                 ^
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:137:18: note: candidate constructor
  /* implicit */ String(double  n);
                 ^
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:146:18: note: candidate constructor has been explicitly deleted
  /* implicit */ String(char) = delete;
                 ^
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:142:3: note: candidate constructor
  String(const String& str) : m_str(str.m_str) { }
  ^
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:147:18: note: candidate constructor has been explicitly deleted
  /* implicit */ String(Variant&&) = delete;
                 ^
/tmp/hhvm-20170206-89135-1140ils/hhvm-3.17.2/hphp/runtime/base/type-string.h:154:18: note: candidate constructor
  /* implicit */ String(String&& str) noexcept : m_str(std::move(str.m_str)) {}
                 ^

**Comment by juanfernandoe** [comment]
+1

**Comment by fredemmott** [comment]
Compile issue appears fixed, hhvm/homebrew-hhvm#99 to edit the brew script
